File: examples/python/basic.py
"""
Base64 Encoder/Decoder API - Basic Usage Example

This example demonstrates the basic usage of the Base64 Encoder/Decoder API.
API Documentation: https://docs.apiverve.com/ref/base64
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/base64'

def call_base64_api():
    """
    Make a GET request to the Base64 Encoder/Decoder API
    """
    try:
        # Query parameters
        params &#x3D; {&#x27;text&#x27;: &#x27;Hello World&#x27;, &#x27;action&#x27;: &#x27;encode&#x27;}

        headers = {
            'x-api-key': API_KEY
        }

        response = requests.get(API_URL, headers=headers, params=params)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Base64 Encoder/Decoder API...\n')

    result = call_base64_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: examples/javascript/basic.js
/**
 * Base64 Encoder/Decoder API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Base64 Encoder/Decoder API.
 * API Documentation: https://docs.apiverve.com/ref/base64
 */

const API_KEY = process.env.APIVERVE_API_KEY || 'YOUR_API_KEY_HERE';
const API_URL = 'https://api.apiverve.com/v1/base64';

/**
 * Make a GET request to the Base64 Encoder/Decoder API
 */
async function callBase64Encoder/DecoderAPI() {
  try {
    // Query parameters
    const params &#x3D; new URLSearchParams({
            text: &#x27;Hello World&#x27;,
            action: &#x27;encode&#x27;
        });

    const response = await fetch(`${API_URL}?${params}`, {
      method: 'GET',
      headers: {
        'x-api-key': API_KEY
      }
    });

    // Check if response is successful
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();

    // Check API response status
    if (data.status === 'ok') {
      console.log('✓ Success!');
      console.log('Response data:', data.data);
      return data.data;
    } else {
      console.error('✗ API Error:', data.error || 'Unknown error');
      return null;
    }

  } catch (error) {
    console.error('✗ Request failed:', error.message);
    return null;
  }
}

// Run the example
callBase64Encoder/DecoderAPI()
  .then(result => {
    if (result) {
      console.log('\n📊 Final Result:');
      console.log(JSON.stringify(result, null, 2));
    }
  });
